assets/js/search/highlightSearchTerm.js

```diff
@@ -10,14 +10,32 @@ window.onload = function () {
     return
   }
 
-  for (const result of searchResults) {
-    result.innerHTML = caseInsensitiveHighlighting(result.innerHTML, searchTerm)
-  }
+  highlightTextInDOM(searchTerm)
 }
 
-function caseInsensitiveHighlighting(html, searchTerm) {
-  const reg = new RegExp(searchTerm, 'gi')
-  return html.replace(reg, function (str) {
-    return `<mark>${str}</mark>`
-  })
+function highlightTextInDOM(targetText) {
+  const walker = document.createTreeWalker(
+    searchResults,
+    NodeFilter.SHOW_TEXT,
+    {
+      acceptNode: function (node) {
+        const formattedNodeValue = node.nodeValue?.toLocaleLowerCase()
+        const formattedTargetText = targetText.toLocaleLowerCase()
+        return formattedNodeValue.includes(formattedTargetText) ? NodeFilter.FILTER_ACCEPT : NodeFilter.FILTER_SKIP
+      },
+    },
+    false
+  )
+  const nodesToUpdate = []
+  let node
+  while ((node = walker.nextNode())) {
+    nodesToUpdate.push(node)
+  }
+
+  for (const textNode of nodesToUpdate) {
+    const span = document.createElement('span')
+    const regEx = new RegExp(targetText, 'ig')
```

Comment:
added this regex so the replaceAll could find target text to replace ignoring text case